### PR TITLE
[MRG+1]Clarify priority adjust settings docs

### LINKED ISSUE #1593: Fix REDIRECT_PRIORITY_ADJUST docs
Not sure about it, but what's explained here: http://scrapy.readthedocs.org/en/latest/topics/settings.html#redirect-priority-adjust about priorities seems wrong, priorities in requests work the other way around (higher priority executes first), so a negative adjust would meant to decrease the original request priority to execute the redirect later.

By all means if this isn't the case, some note about why it seems to contradict what it is said about priorities in http://scrapy.readthedocs.org/en/latest/topics/request-response.html#request-objects would be helpful, got really confused there :(

### PR DESCRIPTION
**Pull request by redapple**
Fixes #1593

### CODE REVIEW

**Comment by kmike** [comment]
what about DEPTH_PRIORITY?

**Comment by redapple** [comment]
@kmike , do you mean also changing the description for http://doc.scrapy.org/en/latest/topics/settings.html#depth-priority ?

**Comment by kmike** [comment]
yes

**Comment by kmike** [comment]
I wonder if we should rename it to DEPTH_PRIORITY_ADJUST to be consistent with other settings. Though adjustment algorithm is different from other settings (which is also not explained in docs): `request.priority -= request.meta['depth'] * DEPTH_PRIORITY_ADJUST` .

**Comment by redapple** [comment]
@kmike , I'd rather we do not change the setting name to `DEPTH_PRIORITY_ADJUST` as it behave differently to the 2 others.
(we could rename to `DEPTH_PRIORITY_NEGATIVE_ADJUST` to be explicit but I wonder if it would not confuse more than anything)

**Comment by kmike** [comment]
@redapple a good point. 
We can make `DEPTH_PRIORITY_ADJUST` option an opposite of `DEPTH_PRIORITY`.

**Comment by redapple** [comment]
@kmike , is renaming ok in a seperate PR?

**Comment by kmike** [comment]
@redapple sure, improved docs is already a big step forward

File: docs/topics/settings.rst [hunk 1]
```diff
@@ -286,9 +288,20 @@ DEPTH_PRIORITY
 
 Default: ``0``
 
-An integer that is used to adjust the request priority based on its depth.
+Scope: ``scrapy.spidermiddlewares.depth.DepthMiddleware``
+
+An integer that is used to adjust the request priority based on its depth:
+
+- **a positive value will decrease the priority**
```
Comment (kmike) [inline docs/topics/settings.rst]: Can we be even more explicit? E.g. "it means requests with larger depth will be processed later".

Comment (kmike) [inline docs/topics/settings.rst]: I always had troubles reading scrapy docs when priorities are explained :) Sometimes a larger value means 'more priority', sometimes it means 'later in a queue', this is confusing.

**Comment by redapple** [comment]
ok @kmike , I also added a link to the FAQ entry about BFO/DFO.

**Comment by kmike** [comment]
:+1:

**Pull request merged**